### PR TITLE
global: new package structure for Invenio 3

### PR DESCRIPTION
**Pull request by omelkonian**
- Cleaning the whole repository to accommodate new version.

Signed-off-by: Orestis Melkonian melkon.or@gmail.com

### CODE REVIEW

**Comment by omelkonian** [comment]
ping @jirikuncar

**Comment by omelkonian** [comment]
Now I have to wait for this to be merged, in order to submit the next PR, right?

**Comment by jirikuncar** [comment]
@omelkonian you can submit WIP for the next PR, just remember to keep it up-to-date.

File: invenio_sequencegenerator/__init__.py [hunk 1]
```diff
@@ -23,3 +23,10 @@
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
 """Invenio module for generating sequences."""
+
+from __future__ import absolute_import, print_function
+
+from .version import __version__
+
+__all__ = ('__version__',
+           )
```
Comment (jirikuncar) [inline invenio_sequencegenerator/__init__.py]: You don't have to break the line here. If you prefer to break it, then in this case it is preferable following style:

``` python
__all__ = (
    '__version__',
)
```

File: invenio_sequencegenerator/version.py [hunk 2]
```diff
@@ -28,4 +28,4 @@
 and parsed by ``setup.py``.
 """
 
-__version__ = "0.1.0.dev20150000"
+__version__ = "0.1.0.dev20160000"
```
Comment (jirikuncar) [inline invenio_sequencegenerator/version.py]: please update a copyright year above too

File: setup.py [hunk 3]
```diff
@@ -25,65 +25,52 @@
...
-        self.pytest_args = []
-        try:
-            from ConfigParser import ConfigParser
-        except ImportError:
-            from configparser import ConfigParser
-        config = ConfigParser()
-        config.read('pytest.ini')
-        self.pytest_args = config.get('pytest', 'addopts').split(' ')
+install_requires = [
+    'Flask-BabelEx>=0.9.2',
```
Comment (jirikuncar) [inline setup.py:74]: IMHO we should include `'Flask>=0.11.1',` too.

File: setup.py [hunk 4]
```diff
@@ -92,30 +79,24 @@ def run_tests(self):
...
-            'bibtex = invenio_sequencegenerator.scripts.bibtex:main',
-        ],
-    },
-    install_requires=requirements,
-    extras_require={
-        'docs': [
-            'Sphinx>=1.3',
-            'sphinx_rtd_theme>=0.1.7'
+        'invenio_i18n.translations': [
+            'messages = invenio_sequencegenerator',
```
Comment (jirikuncar) [inline setup.py:104]: Just make sure that you include catalog in `invenio_sequencegenerator/translations`.

```
mkdir invenio_sequencegenerator/translations
python setup.py extract_messages
python setup.py init_catalog -l fr
```

**Comment by jirikuncar** [comment]
@omelkonian although I like Latin, can you stick to English commit message e.g. `global: new package structure for Invenio 3`? Thanks

File: invenio_sequencegenerator/version.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2015, 2016 CERN.
```
Comment (jirikuncar) [inline invenio_sequencegenerator/version.py:4]: 👍

File: setup.py [hunk 6]
```diff
@@ -92,30 +80,28 @@ def run_tests(self):
...
         ],
-    },
-    install_requires=requirements,
-    extras_require={
-        'docs': [
-            'Sphinx>=1.3',
-            'sphinx_rtd_theme>=0.1.7'
+        'invenio_db.models': [
+            'invenio_pidstore_reportnumber = '
```
Comment (jirikuncar) [inline setup.py]: invenio_pidstore_reportnumber -> invenio_sequencegenerator

Comment (omelkonian) [inline setup.py]: oops!

File: tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    def teardown():
+        with app.app_context():
+            db.session.close()
+            if str(db.engine.url) != 'sqlite://':
+                drop_database(str(db.engine.url))
+            shutil.rmtree(instance_path)
+
+    request.addfinalizer(teardown)
+
+    app.test_request_context().push()
```
Comment (jirikuncar) [inline tests/conftest.py]: please have a look at `@pytest.yield_fixture` and example in https://github.com/inveniosoftware/invenio-records/blob/master/tests/conftest.py#L45-L73

Comment (omelkonian) [inline tests/conftest.py]: :+1:

File: MANIFEST.in [hunk 8]
```diff
@@ -1,9 +1,38 @@
...
+include *.sh
 include *.txt
 include *.yml
-include .dockerignore .editorconfig
+include .dockerignore
+include .editorconfig
+include .tx/config
+prune docs/_build
+
 include .travis.invenio.cfg
```
Comment (jirikuncar) [inline MANIFEST.in]: not needed anymore ...

**Comment by jirikuncar** [comment]
@omelkonian I don't see anywhere the new `.travis.yml`. I'm quite sure that you had it before just it wasn't in the root.

**Comment by omelkonian** [comment]
@jirikuncar Sorry, I kept the old one. Updating...

File: .travis.yml [hunk 9]
```diff
@@ -73,8 +75,9 @@ deploy:
...
+    secure: O5BAgL7uQLyhVGk5/MLVUdDTBtaNexJNWNTXUioW97b6Sl8KdD2s+uJmVIluJ1CtO0r/uvogLNEyrDKPHBNTw8diPN+yXLhCpZe0aiM1AKYmIGDulicQcTNLG176A3QkYc1DW9HMN2d6Tm4bi3kieACtqByACgDPpso8engTvqBmHsl9f8VvEFdoT3O6/yPCi7P7IgWewr2/0l18Y/kmkQPiPYXmBF8hshuP812aV1h0MohxRoYl+rvIlGBFd5F06UZg/1mjxDCZgwxWEzutpZNW3doDZ4cPnJebv4SPuwY8+pTsJ8icIkoPofvkolhcudWLnfaPaR0dVIyHF1OkuNffLFMLKV8VxtHmmThACeRqUpi9Fs4u2ru7q84yDXPnM6cl9koNjAQRsT/caTzL/wNGJ/0SlcuoMEuwDBZWdOAxFf66l/ZyxjQTHLonybUPHKac1lsiV3IcVrJn4Ke4WttmXqGYcV1Rj6sdT7nFVUC0DvWiE+JkBc62XfLSkTWB1Q3rx1Tm7UACwYn6762N3r843EYEM3M4pwgy5IIwIz7XhmCD2W2ZFWXzlvBqBBdpy+w4+uxWk4LbvunNIOXrJtswMnjo93ZIQ2hqdbq3M12B6z74k6h4NjQ7CxhasGdSTCFR6B/5FejFUhH+0tTsUMXUaux7tKg4OaCbP3ZV7Qk=
```
Comment (jirikuncar) [inline .travis.yml]: please make sure that the deploy key is untouched

File: .travis.yml [hunk 10]
```diff
@@ -73,8 +75,9 @@ deploy:
...
   distributions: "compile_catalog sdist bdist_wheel"
   on:
     tags: true
     python: "2.7"
+    condition: $REQUIREMENTS = release
```
Comment (jirikuncar) [inline .travis.yml]: new line is missing

File: docs/conf.py [hunk 11]
```diff
@@ -24,30 +24,28 @@
...
-import sys
 import os
-import shlex
 
-# If extensions (or modules to document with autodoc) are in another directory,
-# add these directories to sys.path here. If the directory is relative to the
-# documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath('..'))
-sys.path.insert(0, os.path.abspath('_ext'))
+import sphinx.environment
```
Comment (jirikuncar) [inline docs/conf.py]: there might be too many empty lines (see failing jobs on travis) after imports

File: setup.py [hunk 12]
```diff
@@ -92,30 +80,28 @@ def run_tests(self):
...
+        'invenio_i18n.translations': [
+            'messages = invenio_sequencegenerator',
         ],
-    },
-    install_requires=requirements,
-    extras_require={
-        'docs': [
-            'Sphinx>=1.3',
-            'sphinx_rtd_theme>=0.1.7'
+        'invenio_db.models': [
```
Comment (jirikuncar) [inline setup.py]: Did you remove the `models.py` file? If it's coming in next PR then please comment or remove this entrypoint. Thanks

**Pull request merged**